manual/src/cmds/runtime-tracing.etex

```diff
@@ -0,0 +1,267 @@
+
+This chapter describes the runtime events tracing system which enables
+continuous extraction of performance information from the OCaml runtime with
+very low overhead. Collected metrics include time spent executing the {\em garbage collector}.
+The overall execution time of individual pauses are measured
+down to the time spent in specific parts of the garbage collection.
+Insight is also given on memory allocation and motion by recording
+the size of allocated memory blocks, as well as value promotions from the
+{\em minor heap} to the {\em major heap}.
+
+\section{s:runtime-tracing-overview}{Overview}
+
+The runtime events tracing system is designed to be used in different contexts:
+\begin{description}
+\item[Self monitoring] OCaml programs and libraries can install their own callbacks
+to listen for runtime events and react to them programmatically, for example,
+to export events to disk or over the network.
+\item[External monitoring] An external process can consume the runtime events
+of an OCaml program whose runtime tracing system has been enabled by setting
+the corresponding environment variable.
+\end{description}
+
+Runtime events functions as a {\em flight recorder} with old events being
+overwritten by new events. This means that consumers can either continuously
+consume events or choose to only do so in response to some circumstance, e.g if
+a particular query or operation takes longer than expected to complete.
+
+There are three main classes of events emitted by runtime events:
+
+\begin{description}
+\item[Spans] Events having a duration in time. An example of a span event in the
+OCaml garbage collector would be a a minor collection. Spans may contain other
+spans, e.g there may be other spans that begin after a minor collection has
+begun and end before it does.
+\item[Lifecycle events] Events that occur at a moment in time. An example of a
+lifecycle event would be the termination of a domain.
+\item[Counters] Events that include the measurement of some quantity of
+interest. An example of a counter would be the number of words promoted from
+the minor to the major heap during the last minor garbage collection.
+\end{description}
+
+\section{s:runtime-tracing-architecture}{Architecture}
+
+The runtime tracing system can be split in to two parts. The probes which emit
+events and the rest of the system that ingest and transport them.
+
+\subsection{s:runtime-tracing-probes}{Probes}
+Probes collect performance events from the runtime system. These are further
+split in to two sets, those always available and those only available in the
+instrumented runtime. Probes in the instrumented runtime are primarily of
+interest to developers of the OCaml runtime and garbage collector and, at
+present, only consist of major heap allocation size counter events.
+
+The full set of events emitted by probes and their documentation can be found in
+\ifouthtml
+ \ahref{libref/Runtime_events.html}{Module \texttt{Runtime_events}}.
```

Comment:
I think it is fine to leave it for further revisions of the manual. Documenting them in the _mli_ sounds like a nice idea.
This could be done in a subsequent PR?
I would be happy to work on this.